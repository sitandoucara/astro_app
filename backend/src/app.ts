import express from "express";
import dotenv from "dotenv";
import cors from "cors";
import chartRoutes from "./routes/chartRoutes";
import timezoneRoutes from "./routes/timezoneRoutes";

dotenv.config();

const app = express();
app.use(cors());
app.use(express.json());

app.use("/api/chart", chartRoutes);
app.use("/api/timezone", timezoneRoutes);

const PORT = process.env.PORT || 4000;
app.listen(PORT, () => {
  console.log(`Server running on port ${PORT}`);
});
