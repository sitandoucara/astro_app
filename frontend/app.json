{
  "expo": {
    "name": "my-expo-app",
    "slug": "my-expo-app",
    "version": "1.0.9",
    "web": {
      "bundler": "metro",
      "favicon": "./assets/favicon.png"
    },
    "experiments": {
      "tsconfigPaths": true
    },
    "plugins": ["expo-font"],
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "userInterfaceStyle": "light",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "assetBundlePatterns": ["**/*"],
    "ios": {
      "supportsTablet": false,
      "bundleIdentifier": "com.sigraph.astromood",
      "infoPlist": {
        "ITSAppUsesNonExemptEncryption": false
      }
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#ffffff"
      }
    }
  }
}
