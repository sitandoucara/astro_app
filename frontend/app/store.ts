import AsyncStorage from '@react-native-async-storage/async-storage';
import { configureStore } from '@reduxjs/toolkit';
import { persistStore, persistReducer } from 'redux-persist';

import authReducer from '../features/auth/AuthSlice';
import languageReducer from '../shared/language/languageSlice';
import themeReducer from '../shared/theme/themeSlice';

const themePersistConfig = {
  key: 'theme',
  storage: AsyncStorage,
};

const languagePersistConfig = {
  key: 'language',
  storage: AsyncStorage,
};

const persistedThemeReducer = persistReducer(themePersistConfig, themeReducer);
const persistedLanguageReducer = persistReducer(languagePersistConfig, languageReducer);

export const store = configureStore({
  reducer: {
    auth: authReducer,
    theme: persistedThemeReducer,
    language: persistedLanguageReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
