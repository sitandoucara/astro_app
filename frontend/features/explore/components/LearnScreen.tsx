import { MaterialIcons } from '@expo/vector-icons';
import Ionicons from '@expo/vector-icons/Ionicons';
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { RootStackParamList } from 'navigation/types';
import { useLayoutEffect } from 'react';
import { Text, TouchableOpacity, View, ScrollView, Alert } from 'react-native';
import Animated, { FadeInUp } from 'react-native-reanimated';
//import { useAppSelector } from 'shared/hooks';
import { useThemeColors } from 'shared/hooks/useThemeColors';

export default function LearnScreen({ onBack }: any) {
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  //const isDarkMode = useAppSelector((state) => state.theme.isDarkMode);
  const colors = useThemeColors();

  const goBack = () => {
    if (onBack) onBack();
    else navigation.goBack();
  };

  useLayoutEffect(() => {
    navigation.setOptions({
      headerShown: true,
      headerTransparent: true,
      headerTitle: '',
      headerStyle: {
        backgroundColor: colors.backgroundColor,
      },
      headerLeft: () => (
        <TouchableOpacity style={{ marginLeft: 16 }} onPress={goBack}>
          <View className="flex-row gap-2">
            <Ionicons name="arrow-back" size={24} style={{ color: colors.textColor }} />
            <Text
              className="text-aref m-l-2 text-left text-xl font-bold"
              style={{ color: colors.textColor }}>
              Learn Astrology
            </Text>
          </View>
        </TouchableOpacity>
      ),
    });
  }, [navigation]);

  const chapters = [
    {
      id: '01',
      title: 'Level 1: Introduction to Astrology',
      duration: '20 sec',
    },
    {
      id: '02',
      title: 'Level 1: The Zodiac Signs',
      duration: '20 sec',
    },
    {
      id: '03',
      title: 'Level 1: The Planets and Their',
      duration: '20 sec',
    },
    {
      id: '04',
      title: 'Level 2: The Ascendant ',
      duration: '',
    },
    {
      id: '05',
      title: 'Level 2: The Four Elements',
      duration: '',
    },
    {
      id: '06',
      title: 'Level 2: Planetary Aspects',
      duration: '',
    },
    {
      id: '07',
      title: 'Level 3: Birth Chart Interpretation',
      duration: '',
    },
    {
      id: '08',
      title: 'Level 3: Synastry and Compatibility',
      duration: '',
    },
    {
      id: '09',
      title: 'Level 3: Planetary Transits',
      duration: '',
    },
    {
      id: '10',
      title: 'Level 3: Retrogrades ',
      duration: '',
    },
  ];

  // Function to determine if a chapter is locked
  const isLocked = (chapter: (typeof chapters)[0]) => {
    const availableLessons = ['01', '02', '03'];
    return !availableLessons.includes(chapter.id);
  };

  // Function to manage clicking on a chapter
  const handleChapterPress = (chapter: (typeof chapters)[0]) => {
    if (isLocked(chapter)) {
      Alert.alert(
        'Coming Soon!',
        `${chapter.title} will be available soon !`,
        [
          {
            text: 'OK',
            style: 'default',
          },
        ],
        { cancelable: true }
      );
    } else {
      navigation.navigate('AudioBookScreen', {
        title: chapter.title,
        jsonUrl: `https://vaajrvpkjbzyqbxiuzsi.supabase.co/storage/v1/object/public/signdetails/learn/lesson_${chapter.id}.json`,
      });
    }
  };

  // Function to get the visual style of locked cards
  const getCardStyle = (chapter: (typeof chapters)[0]) => {
    if (isLocked(chapter)) {
      return {
        opacity: 0.6,
        className: `mt-4 w-full rounded-3xl p-4 ${colors.cardBg} border ${colors.borderColor}`,
      };
    }
    return {
      opacity: 1,
      className: `mt-4 w-full rounded-3xl p-4 ${colors.cardBg} border ${colors.borderColor}`,
    };
  };

  // Function to get the text style according to the lock state
  const getTextStyle = (chapter: (typeof chapters)[0], baseStyle: string) => {
    if (isLocked(chapter)) {
      return `${baseStyle} opacity-60`;
    }
    return baseStyle;
  };

  return (
    <View className="flex-1 p-2" style={{ backgroundColor: colors.backgroundColor }}>
      <ScrollView className="mt-20">
        <View>
          <View className="mt-8">
            {chapters.map((chapter, idx) => {
              const cardStyle = getCardStyle(chapter);
              const locked = isLocked(chapter);

              return (
                <Animated.View key={chapter.id} entering={FadeInUp.delay(idx * 50).duration(400)}>
                  <TouchableOpacity
                    activeOpacity={locked ? 0.7 : 0.8}
                    onPress={() => handleChapterPress(chapter)}
                    className={cardStyle.className}
                    style={{ opacity: cardStyle.opacity }}>
                    <View className="flex-row items-center">
                      <View className="mr-4 w-10 items-center justify-center">
                        {/* Chapter number */}
                        <Text
                          className={getTextStyle(
                            chapter,
                            `text-aref text-lg font-bold ${colors.textPrimary}`
                          )}>
                          {chapter.id}
                        </Text>
                      </View>

                      <View className="flex-1">
                        <Text
                          className={getTextStyle(
                            chapter,
                            `text-aref whitespace-nowrap text-base font-medium ${colors.textPrimary} mb-1`
                          )}>
                          {chapter.title}
                        </Text>
                        <Text
                          className={getTextStyle(
                            chapter,
                            `text-aref whitespace-nowrap text-sm ${colors.textSecondaryAlt}`
                          )}>
                          {chapter.duration}
                        </Text>
                      </View>

                      {/* Padlock  or play icon */}
                      <View className="ml-3 flex-row items-center">
                        {locked ? (
                          <View className="h-10 w-10 items-center justify-center">
                            <MaterialIcons
                              name="lock"
                              size={24}
                              style={{ color: colors.iconColorAlt2, opacity: 0.7 }}
                            />
                          </View>
                        ) : (
                          <View className="h-10 w-10 items-center justify-center">
                            <Ionicons
                              name="play-circle"
                              size={32}
                              style={{ color: colors.iconColorAlt2 }}
                            />
                          </View>
                        )}
                      </View>
                    </View>
                  </TouchableOpacity>
                </Animated.View>
              );
            })}
          </View>
        </View>
      </ScrollView>
    </View>
  );
}
