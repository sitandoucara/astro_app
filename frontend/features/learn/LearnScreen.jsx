import { useLayoutEffect } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { useNavigation } from '@react-navigation/native';

export default function LearnScreen() {
  const navigation = useNavigation();

  useLayoutEffect(() => {
    navigation.setOptions({
      headerTitle: () => (
        <Text
          style={{
            fontFamily: 'ArefRuqaa_400Regular',
            fontSize: 20,
            color: '#32221E',
            marginLeft: 20,
          }}>
          You can learn
        </Text>
      ),
      headerTitleAlign: 'left',
    });
  }, [navigation]);

  return (
    <View style={styles.container}>
      <Text style={styles.welcomeText}>Learn page</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#F2EAE0',
  },
  welcomeText: {
    color: '#7B635A',
    fontFamily: 'ArefRuqaa_400Regular',
    fontWeight: 'bold',
    fontSize: 42,
    marginBottom: 20,
  },
  button: {
    backgroundColor: '#c47b5a',
    borderRadius: 40,
    paddingHorizontal: 20,
    paddingVertical: 10,
    width: 200,
    alignItems: 'center',
    marginVertical: 10,
    overflow: 'hidden',
  },

  buttonText: {
    color: 'white',
    fontWeight: 'bold',
  },
});
