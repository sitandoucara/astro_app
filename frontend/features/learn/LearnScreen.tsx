import { useLayoutEffect } from 'react';
import { useNavigation } from '@react-navigation/native';
import { Text, TouchableOpacity, View, Image, Alert } from 'react-native';
import { useAppSelector } from 'shared/hooks';
import { Feather, MaterialCommunityIcons } from '@expo/vector-icons';

export default function LearnScreen() {
  const isDarkMode = useAppSelector((state) => state.theme.isDarkMode);

  const backgroundColor = isDarkMode ? '#F2EAE0' : '#281109';
  const textColor = isDarkMode ? '#32221E' : '#F2EAE0';
  const titleColor = isDarkMode ? '#7B635A' : '#D8C8B4';
  const navigation = useNavigation();

  // Couleurs dynamiques
  const cardBg = isDarkMode ? 'bg-light-cardback' : 'bg-dark-cardback';
  const borderColor = isDarkMode ? 'border-light-border' : 'border-dark-border';
  const textPrimary = isDarkMode ? 'text-light-text1' : 'text-dark-text1';
  const textSecondary = isDarkMode ? 'text-[#D8D3D0]' : 'text-[#D9D5D4]';

  useLayoutEffect(() => {
    navigation.setOptions({
      headerTitle: () => (
        <Text className="text-aref ml-5 text-[20px]" style={{ color: textColor }}>
          You can learn
        </Text>
      ),
      headerTitleAlign: 'left',
    });
  }, [navigation, isDarkMode]);

  return (
    <View className="flex-1 items-center justify-center px-2" style={{ backgroundColor }}>
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => Alert.alert('Zodiac Sign Compatibility clicked!')}
        className={` w-full rounded-3xl p-6 ${cardBg} border ${borderColor} mb-4`}>
        <View className="  justify-center">
          {/* Header avec les 2 grandes icones */}
          <View className="h-16 w-16 items-center justify-center ">
            <View className="mr-3 h-10 w-10 items-center justify-center rounded-full bg-[#281109] ">
              <MaterialCommunityIcons name="paperclip" size={20} color="#F2EAE0" />
            </View>
          </View>

          {/* Texte centré */}
          <View className="">
            <Text className={`text-aref text-xl font-semibold ${textPrimary} mb-2`}>Learn</Text>
            <Text className={`text-aref text-sm ${textSecondary}`}>
              304 Reports delivered today
            </Text>
          </View>
        </View>
      </TouchableOpacity>

      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => Alert.alert('Zodiac Sign Compatibility clicked!')}
        className={` w-full rounded-3xl p-6 ${cardBg} border ${borderColor} mb-4`}>
        <View className="  justify-center">
          {/* Header avec les 2 grandes icones */}
          <View className="h-16 w-16 items-center justify-center ">
            <View className="mr-3 h-10 w-10 items-center justify-center rounded-full bg-[#281109] ">
              <MaterialCommunityIcons name="book" size={20} color="#F2EAE0" />
            </View>
          </View>

          {/* Texte centré */}
          <View className="">
            <Text className={`text-aref text-xl font-semibold ${textPrimary} mb-2`}>
              Mini-books
            </Text>
            <Text className={`text-aref text-sm ${textSecondary}`}>
              304 Reports delivered today
            </Text>
          </View>
        </View>
      </TouchableOpacity>

      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => Alert.alert('Zodiac Sign Compatibility clicked!')}
        className={` w-full rounded-3xl p-6 ${cardBg} border ${borderColor} mb-4`}>
        <View className="  justify-center">
          {/* Header avec les 2 grandes icones */}
          <View className="h-16 w-16 items-center justify-center ">
            <View className="mr-3 h-10 w-10 items-center justify-center rounded-full bg-[#281109] ">
              <MaterialCommunityIcons name="gamepad-variant" size={20} color="#F2EAE0" />
            </View>
          </View>

          {/* Texte centré */}
          <View className="">
            <Text className={`text-aref text-xl font-semibold ${textPrimary} mb-2`}>
              Test & Quiz
            </Text>
            <Text className={`text-aref text-sm ${textSecondary}`}>
              304 Reports delivered today
            </Text>
          </View>
        </View>
      </TouchableOpacity>
    </View>
  );
}
