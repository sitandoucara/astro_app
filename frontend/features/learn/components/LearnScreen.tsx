import { MaterialIcons } from '@expo/vector-icons';
import Ionicons from '@expo/vector-icons/Ionicons';
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { RootStackParamList } from 'navigation/types';
import { useLayoutEffect } from 'react';
import { Text, TouchableOpacity, View, ScrollView } from 'react-native';
import { useAppSelector } from 'shared/hooks';

export default function LearnScreen({ onBack }: any) {
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  const isDarkMode = useAppSelector((state) => state.theme.isDarkMode);

  const backgroundColor = isDarkMode ? '#F2EAE0' : '#281109';
  const textColor = isDarkMode ? '#32221E' : '#F2EAE0';
  const cardBg = isDarkMode ? 'bg-light-cardback' : 'bg-[#442F29]/50';
  const borderColor = isDarkMode ? 'border-light-border' : 'border-dark-border';
  const textPrimary = isDarkMode ? 'text-light-text1' : 'text-dark-text1';

  const iconColor = isDarkMode ? '#32221E' : '#F2EAE0';
  const textSecondary = isDarkMode ? 'text-[#D8D3D0]' : 'text-[#D9D5D4]';

  const goBack = () => {
    if (onBack) onBack();
    else navigation.goBack();
  };

  useLayoutEffect(() => {
    navigation.setOptions({
      headerShown: true,
      headerTransparent: true,
      headerTitle: '',
      headerLeft: () => (
        <TouchableOpacity style={{ marginLeft: 16 }} onPress={goBack}>
          <View className="flex-row gap-2">
            <Ionicons name="arrow-back" size={24} style={{ color: textColor }} />
            <Text
              className="text-aref m-l-2 text-left text-xl font-bold"
              style={{ color: textColor }}>
              Learn Astrology
            </Text>
          </View>
        </TouchableOpacity>
      ),
    });
  }, [navigation]);

  const chapters = [
    {
      id: '01',
      title: 'Level 1: Introduction to Astrology',
      duration: '16 min 9 sec',
    },
    {
      id: '02',
      title: 'Level 2: The Zodiac Signs',
      duration: '16 min 10 sec',
    },
    {
      id: '03',
      title: 'Level 3: The Planets and Their',
      duration: '20 min 2 sec',
    },
    {
      id: '04',
      title: 'Level 4: The Ascendant and the Houses',
      duration: '23 min 8 sec',
    },
    {
      id: '05',
      title: 'Level 5: The Four Elements',
      duration: '15 min 5 sec',
    },
    {
      id: '06',
      title: 'Level 2: UI/UX Design Tools',
      duration: '19 min 2 sec',
    },
    {
      id: '07',
      title: 'Level 3: UI/UX Design Tools',
      duration: '19 min 2 sec',
    },
    {
      id: '08',
      title: 'Level 3: UI/UX Design Tools',
      duration: '19 min 2 sec',
    },
  ];

  // Fonction pour déterminer si un chapitre est verrouillé
  const isLocked = (title: string) => {
    return title.includes('Level 2') || title.includes('Level 3');
  };

  return (
    <View className="flex-1 justify-center p-2" style={{ backgroundColor }}>
      <View>
        {/* Titre */}

        {/* Liste des chapitres */}
        <View className="mt-8">
          {chapters.map((chapter) => (
            <TouchableOpacity
              key={chapter.id}
              activeOpacity={0.8}
              onPress={() =>
                navigation.navigate('AudioBookScreen', {
                  title: chapter.title,
                  jsonUrl: `https://vaajrvpkjbzyqbxiuzsi.supabase.co/storage/v1/object/public/signdetails/learn/lesson_${chapter.id}.json`,
                })
              }
              className={`mt-4 w-full rounded-3xl p-4 ${cardBg} border ${borderColor}`}>
              <View className="flex-row items-center">
                {/* Numéro du chapitre - centré horizontalement */}
                <View className="mr-4 w-10 items-center justify-center">
                  <Text className={`text-aref text-lg font-bold ${textPrimary}`}>{chapter.id}</Text>
                </View>

                {/* Contenu du chapitre */}
                <View className="flex-1">
                  <Text
                    className={`text-aref whitespace-nowrap text-base font-medium ${textPrimary} mb-1`}>
                    {chapter.title}
                  </Text>
                  <Text className={`text-aref whitespace-nowrap text-sm ${textSecondary}`}>
                    {chapter.duration}
                  </Text>
                </View>

                {/* Bouton play et cadenas */}
                <View className="ml-3 flex-row items-center">
                  {/* Cadenas pour Level 2 et 3 */}
                  {isLocked(chapter.title) && (
                    <View className="ml-2 h-8 w-8 items-center justify-center">
                      <MaterialIcons name="lock" size={20} style={{ color: iconColor }} />
                    </View>
                  )}

                  {/* Icône play */}
                  <View className="h-10 w-10 items-center justify-center">
                    <Ionicons name="play-circle" size={32} style={{ color: iconColor }} />
                  </View>
                </View>
              </View>
            </TouchableOpacity>
          ))}
        </View>
      </View>
    </View>
  );
}
