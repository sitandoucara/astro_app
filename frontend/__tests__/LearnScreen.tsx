import { NavigationContainer } from '@react-navigation/native';
import { render, fireEvent } from '@testing-library/react-native';
import { store } from 'app/store';
import LearnScreen from 'features/learn/LearnScreen';
import React from 'react';
import { Alert } from 'react-native';
import { Provider } from 'react-redux';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      setOptions: jest.fn(),
    }),
  };
});

const renderWithProviders = (component: React.ReactElement) => {
  return render(
    <Provider store={store}>
      <NavigationContainer>{component}</NavigationContainer>
    </Provider>
  );
};

describe('LearnScreen Tests', () => {
  it("A - L'écran Learn s'affiche sans crash", () => {
    const { toJSON } = renderWithProviders(<LearnScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("B - Le bouton 'Learn' est présent", () => {
    const { getByText } = renderWithProviders(<LearnScreen />);
    expect(getByText('Learn')).toBeTruthy();
  });

  it("C - Le bouton 'Mini-books' est présent", () => {
    const { getByText } = renderWithProviders(<LearnScreen />);
    expect(getByText('Mini-books')).toBeTruthy();
  });

  it("D - Le bouton 'Test & Quiz' est présent", () => {
    const { getByText } = renderWithProviders(<LearnScreen />);
    expect(getByText('Test & Quiz')).toBeTruthy();
  });

  it('E - Le bouton déclenche bien une alerte', () => {
    const alertMock = jest.spyOn(Alert, 'alert').mockImplementation(() => {});
    const { getByText } = renderWithProviders(<LearnScreen />);
    fireEvent.press(getByText('Learn'));
    expect(alertMock).toHaveBeenCalled();
    alertMock.mockRestore();
  });
});
