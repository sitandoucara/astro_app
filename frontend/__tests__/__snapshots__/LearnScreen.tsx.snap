// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnScreen Tests A - L'Ã©cran Learn s'affiche sans crash 1`] = `
<View
  className="flex-1 items-center justify-center px-2"
  style={
    {
      "backgroundColor": "#281109",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="  justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View
        className=""
      >
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Learn
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          304 Reports delivered today
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="  justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View
        className=""
      >
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Mini-books
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          304 Reports delivered today
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="  justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View
        className=""
      >
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Test & Quiz
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          304 Reports delivered today
        </Text>
      </View>
    </View>
  </View>
</View>
`;
